sparse/rest/handlers.go

```diff
@@ -0,0 +1,166 @@
+package rest
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/rancher/sparse-tools/sparse"
+)
+
+func (server *SyncServer) sendErrorResponse(errorMessage string, writer http.ResponseWriter) {
+	log.Error(errorMessage)
+	http.Error(writer, errorMessage, http.StatusInternalServerError)
+}
+
+func (server *SyncServer) getQueryInterval(request *http.Request) (sparse.Interval, error) {
+	var interval sparse.Interval
+	var err error
+
+	queryParams := request.URL.Query()
+	if len(queryParams["begin"]) == 0 || len(queryParams["end"]) == 0 {
+		err = fmt.Errorf("queryParams begin or end not exist")
+		return interval, err
+	}
+
+	begin, err := strconv.ParseInt(queryParams["begin"][0], 10, 64)
+	if err != nil {
+		err = fmt.Errorf("strconv.ParseInt(begin) error: %s", err)
+		return interval, err
+	}
+	end, err := strconv.ParseInt(queryParams["end"][0], 10, 64)
+	if err != nil {
+		err = fmt.Errorf("strconv.ParseInt(end) error: %s", err)
+		return interval, err
+	}
+
+	return sparse.Interval{begin, end}, err
+}
+
+func (server *SyncServer) open(writer http.ResponseWriter, request *http.Request) {
+	// get file size
+	interval, err := server.getQueryInterval(request)
+	if err != nil {
+		errorMessage := fmt.Sprintf("server.getQueryInterval failed, err: %s", err)
```

Comment:
Looks a little cumbersome, but it gives me info about at which call it fails. Here I know it fails at getQueryInterval. If using a wrapper function, then you get error info at top level function open(), ie. if you get a "Invalid Parameter" error, sometimes it is not obvious to know which sub-function call fails. In our case, you don't know if it is getQueryInterval() or NewDirectFileIoProcessor() or Truncate() fails.